### PR TITLE
Update default Ticks Logic

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

Updated the default logic for ticks with 2 axes.

### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import LOCALE, { DEFAULT_TICK_FORMAT } from '../locale';
 import { prepareHAxis } from './datetimeBuckets';
 import { shouldTruncateLabel, truncateLabel } from './label';
 import * as localeFormat from './localeFormat';
+// import { cosh } from 'core-js/core/number';
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:13]: Is this still needed?

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 2]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
     ) {
-      const yTickValues = generateYAxesTickValues(
-        config.axis.y.domain.lowerLimit,
-        config.axis.y.domain.upperLimit,
+      yTickValues = generateYAxesTickValues(
+        yLowerLimit,
+        yUpperLimit,
         config.ticksCount,
-        config.allowCalibration,
         config.axis.y,
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:537]: `generateYAxesTickValues` now takes only three parameters. Does `config.axis.y` still need to be passed?

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 3]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
       return axis;
     }
     // Y and Y2 axes - ticksCount.
+    let yLowerLimit; let yUpperLimit; let y2LowerLimit; let
+      y2UpperLimit;
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:505]:
```suggestion
    let yLowerLimit;
    let yUpperLimit;
    let y2LowerLimit;
    let y2UpperLimit;
```

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 4]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
+        .domain([yLowerLimit, yUpperLimit])
+        .nice()
+        .domain();
+      [y2LowerLimit, y2UpperLimit] = d3
+        .scaleLinear()
+        .domain([y2LowerLimit, y2UpperLimit])
+        .nice()
+        .domain();
+    }
+    // console.log(yLowerLimit, yUpperLimit, y2LowerLimit, y2UpperLimit);
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:524]: Remove

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 5]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
+
+    let yTickValues; let
+      y2TickValues;
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:527]:
```suggestion
    let yTickValues;
    let y2TickValues;
```

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 6]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
 
-      const yTickValues = generateYAxesTickValues(
-        config.axis.y.domain.lowerLimit,
-        config.axis.y.domain.upperLimit,
-        ticksCount,
-        config.allowCalibration,
+      let intervalCount;
+
+      // Divide the larger range value (a) by the smaller value (b) and determine if a whole number (c) is returned.
+      // If the return value is a whole number and does not have a value grter than 10, divide the smaller range value (b) by the return value (c)
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:556]:
```suggestion
      // If the return value is a whole number and does not have a value greater than 10, divide the smaller range value (b) by the return value (c)
```

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 7]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
+      // and determine if a whole number is returned. If a whole number is returned, use the initial return value to configure the tick count of the graph.
+      const yRange = yUpperLimit - yLowerLimit;
+      const y2Range = y2UpperLimit - y2LowerLimit;
+      const greaterRange = yRange >= y2Range ? yRange : y2Range;
+      const lowerRange = yRange <= y2Range ? yRange : y2Range;
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:561]: You could use Math.min() and Math.max().

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 8]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
+      const factor = greaterRange / lowerRange;
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:562]: Is there any chance `lowerRange` is 0?

File: packages/carbon-graphs/src/js/helpers/axis.js [hunk 9]
```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
...
+        y2LowerLimit,
+        y2UpperLimit,
+        intervalCount - 1,
         config.axis.y2,
       );
+    }
+    if (
+      config.axis.y.suppressTrailingZeros
+        && utils.isUndefined(config.axis.y.ticks.format)
+    ) {
```
Comment (benbcai) [inline packages/carbon-graphs/src/js/helpers/axis.js:601]: Some of these `if` formats are inconsistent with the rest

Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js:603]: Some of they were automatically updated by the linter. I can take a look at them.

File: packages/carbon-graphs/src/js/helpers/axis.js (hunk 8 above)
Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js:564]: If a consumer deliberately uses it but it doesn't break anything. It'll still render just fine.

File: packages/carbon-graphs/src/js/helpers/axis.js (hunk 1 above)
Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js]: Removed here: c70a5e131bc72a698715c2ff2ac0023705d22a46

File: packages/carbon-graphs/src/js/helpers/axis.js (hunk 4 above)
Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js]: Removed here: c70a5e131bc72a698715c2ff2ac0023705d22a46

File: packages/carbon-graphs/src/js/helpers/axis.js (hunk 7 above)
Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js]: Updated here: c70a5e131bc72a698715c2ff2ac0023705d22a46

File: packages/carbon-graphs/src/js/helpers/axis.js (hunk 2 above)
Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js]: Fixed here: df7d88e6692cb962dbe2020077c85df5b4b27aa6

**Review by jmsv6d — APPROVED** [review]

File: packages/carbon-graphs/src/js/helpers/axis.js (hunk 9 above)
Comment (sdadn) [reply packages/carbon-graphs/src/js/helpers/axis.js:600]: Updated here: 0b0f666a6b1567b0fe02134c22c047f146fe9ca7

**Review by BenBoersma — APPROVED** [review]

**Pull request merged**